### PR TITLE
Makes non-weapon items poke people on help intent instead of hitting them.

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes it so when you're holding an item that isn't meant for hurting others you'll instead poke whomever you're clicking on instead of hitting them. Other intents work the same and weapon items still hurt on help intent.
Also makes pacificsts **always** poke others with items instead of preventing the action alltogether.
<details>
<summary>List of items that are considered weapons : </summary>

All melee weapons.
All guns.
All reagent holders.
Traitor uplink weaponry and items.
Cuffs.
Fireaxe.
C4 and X4.
Plushies.
Glass shards.
Xenoarch artefacts.
Books.
Protokinetic Crusher.
Chairs.
</details>

## Why It's Good For The Game
Accidental missclicks leading to harm are the bane of new players everywhere, and this somewhat helps with that. This also allows fore more RP opportunities where you poke people with something to get their attention. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Normal human using non-harmful and harmful items : 

https://user-images.githubusercontent.com/110184118/214395522-d8d7a9fc-484b-467f-ac80-d63b0949a220.mp4

Pacifist using harmful and non-harumful items : 

https://user-images.githubusercontent.com/110184118/214395567-47a96275-76cf-4c20-bdff-932f758d6370.mp4

Attack logs :

image

Item description of non-weapon items, suggesting that you'll need to switch off harm to actually bludgeon someone with it.

image

</details>

## Changelog
:cl:
tweak: Pacifists now always non-harmfully poke people instead of being unable to hit others at all
tweak: Non-weapon items now non-harmfully poke the living things you click on help intent
add: added a line of text to the description of items that will poke people on help intent
/:cl:

### CODE REVIEW

**Comment by CydiaLamiales** [comment]
This'll definitely need a testmerge or two to ensure that I didn't mess up anything and forgot to add items to the weapons category.
I apologize for all the ruined combat moments in advance.

**Comment by CydiaLamiales** [comment]
Toolboxes listed as weapons due to popular request.

**Comment by TerryTehCoder** [comment]
It was as if 1,000 assistants simultaneously cried out in pain... "how do I switch hands?"

**Comment by bluezorua** [comment]
is the slapping hand of *slap a weapon?

**Comment by CydiaLamiales** [comment]
> is the slapping hand of *slap a weapon?

Basically, yeah.

**Comment by CydiaLamiales** [comment]
> > Non-weapon items now non-harmfully poke the living things you click on harm intent
> 
> > on harm intent
> 
> i can get down with making things not hurt on _help_ intent but on harm intent evrything should hurt

Ah dang that's a typo, meant to say 'help' there
Good catch

**Comment by PowerfulBacon** [comment]
Not entirely sure about this one/convinced on this behaviour. It might help the first click not be an accidental hit, but it could lead to situations where you want to hit someone but can't.
Another potential flaw is new players just playing on one of the harm intents constantly as oppossed to doing combat on help intent, which also defeats the intent of the PR.
This is the kind of thing that needs testing, would be nice if there was some method of determining if someone was trying to attack someone and logging it so we can see which method resulted in less misactions (If they attack once but stop, log it as unintentional and a misaction, if they poke but then start attacking it was probably intentional and is a misaction.).

**Comment by PowerfulBacon** [comment]
Testmerging, will watch chat for comments on this

**Comment by benbot16** [comment]
Hitting things while on help still plays the attack animation, which has the potential to confuse people.

**Comment by benbot16** [comment]
It'd also be nice if this had a pref or some other sort of toggle, since there's definitely times I've wanted to be on help but also whacking people with improvised weapons like welders.

**Comment by benbot16** [comment]
Surgical tools prod people on harm intent.

**Comment by Harranhall** [comment]
How will this affect blob rounds? Currently, players are encouraged to stay on help intent to allow people to rotate out of combat in tight spaces.

**Comment by itsmeow** [comment]
Seems people like it for surgery the most

image
image
image
image

**Review by PowerfulBacon — DISMISSED** [review]
Can you confirm this isn't the cause of the brute pack bug?

**Comment by CydiaLamiales** [comment]
It's not, but don't merge yet.

**Review by itsmeow — APPROVED** [review]
I found this overall fine to use. Usually if someone intends to hurt someone they are on harm, but having weapons override this is good! Seems to be godsend for surgery as well.

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -160,5 +174,34 @@
...
+		poke_message_local = "You [message_verb] yourself with [I]!"
+	visible_message("<span class='notice'>[poke_message]</span>",\
+	"<span class='usernotice'>[poke_message_local]</span>", null, COMBAT_MESSAGE_RANGE)
+
+/mob/living/proc/record_accidental_poking()
+	if(time_of_last_poke != 0 && world.time - time_of_last_poke <= 50)
+		SSblackbox.record_feedback("tally", "poking_data", 1, "Hit someone shortly after poking them")
+
+/mob/living/proc/check_for_accidental_attack()
+	addtimer(CALLBACK(src, .proc/record_accidental_attack, time_of_last_attack_dealt), 100, TIMER_OVERRIDE|TIMER_UNIQUE)
```
Comment (CydiaLamiales) [inline code/_onclick/item_attack.dm]:
```suggestion
	addtimer(CALLBACK(src, PROC_REF(record_accidental_attack), time_of_last_attack_dealt), 100, TIMER_OVERRIDE|TIMER_UNIQUE)
```

**Pull request merged**